### DISCUSSION TITLE
Incorporating external datasets

### DISCUSSION THREAD

**Discussion by riziles** [gorules/zen#114] [Q&A] [answered]
Is there any way to pull in external data for use in decisioning? I tried to write a `fetch` inside a Javascript function, but it did not work.

Use case:

Consider the [Fintech: Company Analysis](https://gorules.io/docs/user-manual/getting-started/tutorials/fintech-company-analysis) example. Suppose that I subscribe to a third-party service where I can check to see if the company or key individuals is in a database of know money launderers and terrorist financiers, and I can call it by sending a request to REST API. Is there a way to incorporate that into the Zen rules engine?

**Comment by stefan-gorules** [comment, marked as answer]
Currently this is not possible directly within the decision. It will be possible once we implement Custom node to handle such scenarios, which is coming soon in Q1 2024.

The best way to handle it for now is to keep it in your application logic and add red flag based on it.

**Comment by riziles** [comment]
Cool! Thanks for the quick response.
